app/lib/features/spaces/pages/create_space_page.dart

```diff
@@ -176,6 +211,76 @@ class _CreateSpacePageConsumerState extends ConsumerState<CreateSpacePage> {
     );
   }
 
+  Widget _buildVisibility() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).visibilityTitle,
+          style: Theme.of(context).textTheme.bodyMedium,
+        ),
+        Text(
+          L10n.of(context).visibilitySubtitle,
+          style: Theme.of(context).textTheme.bodySmall!.copyWith(
+                color: Theme.of(context).colorScheme.neutral4,
+              ),
+        ),
+        const SizedBox(height: 10),
+        InkWell(
+          onTap: () async {
+            final spaceVisibility = ref.read(_selectedVisibilityProvider);
+            final selected = await selectVisibilityDrawer(
+              context: context,
+              selectedVisibilityEnum: spaceVisibility,
+              isLimitedVisibilityShow:
+                  ref.read(selectedSpaceIdProvider) != null,
+            );
+            if (selected != null) {
+              ref
+                  .read(_selectedVisibilityProvider.notifier)
+                  .update((state) => selected);
+            }
+          },
+          child: selectedVisibility(),
+        ),
+      ],
+    );
+  }
+
+  Widget selectedVisibility() {
+    final selectedVisibility = ref.watch(_selectedVisibilityProvider);
+    switch (selectedVisibility) {
+      case RoomVisibility.Public:
+        return RoomVisibilityItem(
+          iconData: Icons.language,
+          title: L10n.of(context).public,
+          subtitle: L10n.of(context).publicVisibilitySubtitle,
+          isShowRadio: false,
+        );
+      case RoomVisibility.Private:
+        return RoomVisibilityItem(
+          iconData: Icons.lock,
+          title: L10n.of(context).private,
+          subtitle: L10n.of(context).privateVisibilitySubtitle,
+          isShowRadio: false,
+        );
+      case RoomVisibility.SpaceVisible:
+        return RoomVisibilityItem(
+          iconData: Atlas.users,
+          title: L10n.of(context).limited,
+          subtitle: L10n.of(context).limitedVisibilitySubtitle,
+          isShowRadio: false,
+        );
+      default:
+        return RoomVisibilityItem(
+          iconData: Icons.lock,
+          title: L10n.of(context).private,
+          subtitle: L10n.of(context).privateVisibilitySubtitle,
+          isShowRadio: false,
+        );
```

Comment:
as said above we probably want to show an error here as there are some `knock_...` we do not yet support...